lib/reversetunnel/conn.go

```diff
@@ -251,3 +254,76 @@ func (c *remoteConn) sendDiscoveryRequest(req discoveryRequest) error {
 
 	return nil
 }
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// emitConn is a wrapper for a net.Conn that emits an audit event for non-Teleport connections.
+type emitConn struct {
+	net.Conn
+	mu       sync.RWMutex
+	buffer   bytes.Buffer
+	emitter  apievents.Emitter
+	ctx      context.Context
+	serverID string
+	emitted  bool
+}
+
+func newEmitConn(ctx context.Context, conn net.Conn, emitter apievents.Emitter, serverID string) *emitConn {
+	return &emitConn{
+		Conn:     conn,
+		emitter:  emitter,
+		ctx:      ctx,
+		serverID: serverID,
+	}
+}
+
+func (conn *emitConn) Read(p []byte) (int, error) {
+	// First bytes of a Teleport connection.
+	const emitConnTargetPrefix = "Teleport"
```

Comment:
What guarantees that these are the first bytes of a Teleport connection?

Adding to that, what guarantees that this won't change, because if it does change then emitConn will silently stop emitting events - we could have a regression and not even notice.